gpt_engineer/core/project_config.py

```diff
@@ -0,0 +1,159 @@
+"""
+Functions for reading and writing the `gpt-engineer.toml` configuration file.
+
+The `gpt-engineer.toml` file is a TOML file that contains project-specific configuration used by the GPT Engineer CLI and gptengineer.app.
+"""
+
+from dataclasses import asdict, dataclass, field
+from pathlib import Path
+
+import tomlkit
+
+default_config_filename = "gpt-engineer.toml"
+
+example_config = """
+[project]
+base_dir = "./frontend"  # base directory to operate in (for monorepos)
+src_dir = "./src"        # source directory (under the base directory) from which context will be retrieved
```

Comment:
I'm not sure about this section. Would perhaps make more sense with a `paths` section, such that it can be just:

```suggestion
[paths]
base = "./frontend"  # base directory to operate in (for monorepos)
src = "./src"        # source directory (under the base directory) from which context will be retrieved
```